src/Test/StateMachine.hs

```diff
@@ -279,7 +279,7 @@ runParallelProgram'
 runParallelProgram' n StateMachine {..} prog =
   replicateM n $ do
     hist <- run (executeParallelProgram semantics' prog)
-    return (hist, linearise transition' postcondition' model' hist)
+    return (hist, linearise transition' (okPostcondition postcondition') model' hist)
```

Comment:
Oh, isn't this screwing up `linearise`? It doesn't seem to be that difficult to change it to do the proper things for `postcondition`s that can react to `Fail`.

Reply:
Fixed, thanks!